### PR TITLE
docs: add howto on uploading with progress from craft-cli (CRAFT-736)

### PR DESCRIPTION
**Pull request by sergiusens**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: craft_store/base_client.py [hunk 1]
```diff
@@ -186,10 +186,23 @@ def upload_file(
         """Upload filepath to storage.
 
         The monitor_callback is a method receiving one argument of type
-        ``MultipartEncoderMonitor``, among others, it's ``.bytes_read`` and
-        ``.len`` attributes can be used to track progress.
+        ``MultipartEncoder``, this method is to return a method to track
+        progress. Among others, it's ``.bytes_read`` and ``.len`` attributes
```
Comment (cmatsuoka) [inline craft_store/base_client.py:190]: This sounds a bit confusing, should that "it's" be "its"? And what object is supposed to have these attributes?

File: craft_store/base_client.py [hunk 2]
```diff
@@ -186,10 +186,23 @@ def upload_file(
...
+        can be used to track progress. The simplest implementation can look like:
+
+        .. code-block:: python
+
+          def monitor_callback(encoder: requests_toolbelt.MultipartEncoder):
+
+              # instantiate progress class with total bytes encoder.len
+
+              def progress_printer(monitor: requests_toolbelt.MultipartEncoderMonitor):
+                 # Print progress using monitor.bytes_read
```
Comment (cmatsuoka) [inline craft_store/base_client.py:200]: Ah, it's the argument to the method returned by `monitor_callback`. So maybe we could say something like: "...return a method that receives an argument of type ``MultipartEncoderMonitor``. This argument contains the ``.bytes_read`` and ``.len`` attributes that are used to track progress"?

**Review by cmatsuoka — COMMENTED** [review]
The howto looks good, but the code comment can be improved for clarity.

File: craft_store/base_client.py (hunk 2 above)
Comment (sergiusens) [reply craft_store/base_client.py:206]: updated with class information for clarity

**Review by cmatsuoka — APPROVED** [review]

**Pull request merged**